### PR TITLE
Cleanup the doxygen documentation

### PR DESCRIPTION
**Pull request by jamiebramwell**
This unifies the doxygen style as javadocs and adds the missing documentation fields. A preview of the generated documentation is available [here](https://serac.readthedocs.io/en/docs-bramwell-doxygen_cleanup/doxygen/html/).

### CODE REVIEW

File: src/solvers/thermal_operators.hpp [hunk 1]
```diff
@@ -15,119 +21,135 @@
...
   std::vector<serac::BoundaryCondition>& ess_bdr_;
 
   /**
-   * Auxillary working vectors
+   * @brief Auxillary working vectors
    */
-  mutable mfem::Vector z_;
-  mutable mfem::Vector y_;
-  mutable mfem::Vector x_;
+  mutable mfem::Vector z_, y_, x_;
```
Comment (joshessman-llnl) [inline src/solvers/thermal_operators.hpp]: ISOCPP [generally advises](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-name-one) to keep one declaration per line, though I'm not especially bothered by the use of the comma operator here.

File: src/coefficients/stdfunction_coefficient.hpp [hunk 2]
```diff
@@ -27,53 +28,79 @@ namespace serac {
...
+  /**
+   * @brief The function to evaluate for the coefficient
+   */
   std::function<double(mfem::Vector&)> func_;
 };
 
 /**
  * @brief StdFunctionVectorCoefficient is an easy way to make an
- * mfem::Coefficient using a lambda
+ * mfem::VectorCoefficient using a lambda
```
Comment (joshessman-llnl) [inline src/coefficients/stdfunction_coefficient.hpp]: :+1:

File: src/integrators/hyperelastic_traction_integrator.hpp [hunk 3]
```diff
@@ -4,26 +4,72 @@
...
+   * @brief Working matrices for the traction calculations
+   */
   mutable mfem::DenseMatrix DSh_u_, DS_u_, J0i_, F_, Finv_, FinvT_, PMatI_u_;
-  mutable mfem::Vector      shape_, nor_, fnor_, Sh_p_, Sh_u_;
 
- public:
+  /**
+   * @brief Working vectors for the traction
+   */
+  mutable mfem::Vector shape_, nor_, fnor_, Sh_p_, Sh_u_;
```
Comment (joshessman-llnl) [inline src/integrators/hyperelastic_traction_integrator.hpp]: This could also be split into separate lines, though maybe small cleanup things like that should be in a separate PR.

Mostly unrelated, but were these working vectors/matrices declared in class scope to avoid repeated constructor calls?

**Review by joshessman-llnl — APPROVED** [review]
Thanks for slogging through this!  We probably don't need to start a formal checklist just yet, but to maintain this level of documentation we need to make sure that all PRs from here on out have Doxygen comments for new/updated functions, classes, etc

File: src/solvers/thermal_operators.hpp (hunk 1 above)
Comment (jamiebramwell) [reply src/solvers/thermal_operators.hpp]: I'm OK either way. It's probably best to keep them on separate lines.

File: src/integrators/hyperelastic_traction_integrator.hpp (hunk 3 above)
Comment (jamiebramwell) [reply src/integrators/hyperelastic_traction_integrator.hpp]: Yep, we didn't want to construct and destruct these dense operators in hot loops.

File: src/coefficients/stdfunction_coefficient.hpp [hunk 4]
```diff
@@ -20,219 +21,288 @@
...
+   * @param[in] s1 A scalar Coefficient to apply Func to
+   * @param[in] s2 A scalar Coefficient to apply Func to
+   * @param[in] func A function that takes in two input scalars, and returns the
+   * output.
+   */
   TransformedScalarCoefficient(std::shared_ptr<mfem::Coefficient> s1, std::shared_ptr<mfem::Coefficient> s2,
                                std::function<double(const double, const double)> func);
 
+  /**
+   * @brief Evalate the coefficient at a quadrature point
```
Comment (white238) [inline src/coefficients/stdfunction_coefficient.hpp]: spelling.

File: src/docs/sphinx/style_guide.rst [hunk 5]
```diff
@@ -42,10 +42,9 @@ can come in handy to provide additional information:
      * Calculate du_dt = M^-1 (-Ku + f).
      * This is all that is needed for explicit methods
      * @param[in] u The state vector (input to the differentiation)
-     * @param[out] du The derivative of \p u with respect to time
+     * @param[out] du The derivative of @p u with respect to time
```
Comment (white238) [inline src/docs/sphinx/style_guide.rst:45]: was this change intentional?

File: src/integrators/inc_hyperelastic_integrator.hpp [hunk 6]
```diff
@@ -4,47 +4,81 @@
 //
 // SPDX-License-Identifier: (BSD-3-Clause)
 
+/**
+ * @file inc_hyperelastic_integrator.hpp
+ *
+ * @brief The MFEM integrators for the incremetal hyperelastic formulation
```
Comment (white238) [inline src/integrators/inc_hyperelastic_integrator.hpp]: incremental

File: src/docs/sphinx/style_guide.rst (hunk 5 above)
Comment (joshessman-llnl) [reply src/docs/sphinx/style_guide.rst:45]: It's a style change - using the `@` instead of the `\` to indicate Doxygen commands

Comment (jamiebramwell) [reply src/docs/sphinx/style_guide.rst:45]: This was probably a find/replace boo boo.

**Comment by jamiebramwell** [comment]
LGTM

**Comment by jamiebramwell** [comment]
> 
> 
> Thanks for slogging through this! We probably don't need to start a formal checklist just yet, but to maintain this level of documentation we need to make sure that all PRs from here on out have Doxygen comments for new/updated functions, classes, etc

The vscode doxygen plugin helped a bunch!

**Review by white238 — APPROVED** [review]

**Comment by jamiebramwell** [comment]
LGTM
